File: CountLines/count_lines_2.pl
#!/bin/env perl

use strict;
use warnings;

# http://learn.perl.org/faq/perlfaq5.html#How-do-I-count-the-number-of-lines-in-a-file
sub count_lines {
  my $filename = shift;

  open( FILE, $filename );
  my $lines = 0;
  $lines++ while <FILE>;
  close FILE;

  return $lines;
}

print count_lines( $ARGV[ 0 ] );

File: CountLines/count_lines_2.rb
#!/usr/bin/env ruby
# encoding: utf-8

def count_lines( filename )
  count = 0
  File.open( filename, 'r' ) do | file |
    count += 1 while file.gets
  end

  return count
end

if __FILE__ == $0
  puts count_lines( ARGV[ 0 ] );
end